ArduCopter/GCS_Mavlink.cpp

```diff
@@ -1033,7 +1033,12 @@ void GCS_MAVLINK_Copter::handleMessage(mavlink_message_t* msg)
                 pos_vector += copter.inertial_nav.get_position();
             } else {
                 // convert from alt-above-home to alt-above-ekf-origin
-                pos_vector.z = copter.pv_alt_above_origin(pos_vector.z);
```

Comment:
I've got to say this was actually incorrect because this message is for local frame, which means it should be an altitude above origin and not above home.

We've actually debated this before and it's my fault this isn't corrected along with LOCAL_POSITION_NED (see https://github.com/ArduPilot/ardupilot/issues/6830 and https://github.com/ArduPilot/ardupilot/pull/6893 for reminders).

I'll recommend to merge as is and I'll make a PR to fix the things above as we've agreed before.